### PR TITLE
Alps 574 flatview

### PR DESCRIPTION
**Pull request by ygilquin-gpsw**
Branch tested with different samples + dev/project/flatview

### CODE REVIEW

**Comment by sw-team-release-gpsw** [comment]
[CLA assistant check](https://cla.gopro.com/sign) <br/>All committers have signed the CLA.

**Comment by bbaudel-gpsw** [comment]
Issue : when reaching a flat view directly the default max vertical fov is correctly set. But if the previous view was an rectilinear with a equi for instance, the max vertical fov value is not the same and the media can be panned. Issue is present for if the flat view is applied on a media image and the previous media was an image too. Same for video to video media.

**Comment by bbaudel-gpsw** [comment]
Not implemented : a flat media without flat view associated. It result a black screen.
See : RenderManager line 793 + ViewManager line 83

File: src/render/shader/ShaderChunk/common/defines.glsl [hunk 1]
```diff
@@ -1,6 +1,10 @@
 precision highp float;
 precision highp int;
 
+#define PI_2  1.5707963268
```
Comment (aberthet-gpsw) [inline src/render/shader/ShaderChunk/common/defines.glsl]: Not used, maybe remove it ?

**Comment by rroux-gpsw** [comment]
I try to use fx sets / post processing on flat view with no success.

File: src/render/shader/ShaderChunk/common/defines.glsl (hunk 1 above)
Comment (ygilquin-gpsw) [inline src/render/shader/ShaderChunk/common/defines.glsl]: Not la peine, it's a constante that could resservir plus late

File: src/render/view/ViewFlat.js [hunk 2]
```diff
@@ -172,3 +172,45 @@ FORGE.ViewFlat.prototype.destroy = function()
     FORGE.ViewBase.prototype.destroy.call(this);
 };
 
+/**
+ * Get and set the repeat X beahvior.
+ * @name  FORGE.ViewFlat#repeatX
+ * @type {string}
+ */
+Object.defineProperty(FORGE.ViewFlat.prototype, "repeatX",
```
Comment (ygilquin-gpsw) [inline src/render/view/ViewFlat.js]: Ouch, it's a bug: it's commited at start once for all for the moment. Please log a bug.

**Comment by bbaudel-gpsw** [comment]
I open tickets for untreated cases and I validate the PR.

**Pull request merged**